### PR TITLE
-Adjusted CIMC proxy module to properly handle HTTP error codes and

### PR DESCRIPTION
**Pull request by spenceation**
### What does this PR do?
Adjusted CIMC proxy module to properly handle HTTP errors and timeouts.

### What issues does this PR fix or reference?
Currently CIMC modules will display error messages when the CIMC proxy timeouts or receives a HTTP error code. This patch will apply the appropriate exception code during proxy module calls.

### Previous Behavior
Error messages are displayed instead of device is not reachable.

### New Behavior
Devices will now properly display not reachable messages when timeouts or HTTP error codes occur.

### Tests written?
No
### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/proxy/cimc.py [hunk 1]
```diff
@@ -150,7 +150,15 @@ def set_config_modify(dn=None, inconfig=None, hierarchical=False):
...
                                 verify_ssl=False,
                                 raise_error=True,
+                                status=True,
                                 headers=DETAILS['headers'])
+
+    if str(r['status']) not in ['200', '201', '202', '204']:
+        log.debug("Received error HTTP status code: {0}" .format(str(r['status'])))
+        logout(cookie)
+        raise salt.exceptions.CommandExecutionError(
+            "Did not receive a valid response from host.")
```
Comment (cachedout) [inline salt/proxy/cimc.py]: Maybe log what the error code we received was?

File: salt/proxy/cimc.py [hunk 2]
```diff
@@ -179,11 +187,18 @@ def get_config_resolver_class(cid=None, hierarchical=False):
                                 decode=True,
                                 verify_ssl=False,
                                 raise_error=True,
+                                status=True,
                                 headers=DETAILS['headers'])
 
+    if str(r['status']) not in ['200', '201', '202', '204']:
```
Comment (cachedout) [inline salt/proxy/cimc.py]: This is identical to L156. Refactor for DRY?

File: salt/proxy/cimc.py [hunk 3]
```diff
@@ -202,7 +217,13 @@ def logon():
                                 decode=True,
                                 verify_ssl=False,
                                 raise_error=False,
+                                status=True,
                                 headers=DETAILS['headers'])
+
+    if str(r['status']) not in ['200', '201', '202', '204']:
```
Comment (cachedout) [inline salt/proxy/cimc.py]: Same comment as above for refactoring for DRY.

File: salt/proxy/cimc.py (hunk 1 above)
Comment (spenceation) [reply salt/proxy/cimc.py]: It should be logging the error code on line 157.

File: salt/proxy/cimc.py (hunk 2 above)
Comment (spenceation) [reply salt/proxy/cimc.py]: I'll change this to be included in a single function.

**Comment by rallytime** [comment]
Hey @spenceation! I'm checking in here - did you get a chance to implement the feedback above?

**Comment by spenceation** [comment]
Hey @rallytime. I've added the requested changes.

File: salt/proxy/cimc.py [hunk 4]
```diff
@@ -97,6 +97,16 @@ def __virtual__():
     return __virtualname__
 
 
+def _validate_response_code(response_code_to_check, cookie_to_logout=None):
+    formatted_response_code = str(response_code_to_check)
+    if formatted_response_code not in ['200', '201', '202', '204']:
+        if cookie_to_logout:
+            logout(cookie_to_logout)
+        log.debug("Received error HTTP status code: {0}" .format(formatted_response_code))
```
Comment (mirceaulinic) [inline salt/proxy/cimc.py]: Perhaps `log.error` would be more appropriate?

**Review by mirceaulinic — APPROVED** [review]
Just a small suggestion, otherwise looks good to me.

File: salt/proxy/cimc.py (hunk 4 above)
Comment (rallytime) [reply salt/proxy/cimc.py]: @spenceation What do you think about this comment here?

**Review by rallytime — APPROVED** [review]

**Pull request merged**